flexget/utils/serialization.py

```diff
@@ -0,0 +1,145 @@
+import datetime
+from abc import ABC, abstractmethod
+
+from flexget.utils import json
+
+
+DATE_FMT = '%Y-%m-%d'
+ISO8601_FMT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+def serialize(value):
+    """
+    Convert an object to JSON serializable format.
+
+    """
+    if isinstance(value, Serializable):
+        return value.serialize()
+    if isinstance(value, datetime.datetime):
+        return DateTimeSerializer.serialize(value)
+    if isinstance(value, datetime.date):
+        return DateSerializer.serialize(value)
+    elif isinstance(value, (str, int, float, dict, list)):
+        return BuiltinSerializer.serialize(value)
+    else:
+        raise TypeError('%r is not serializable', value)
+
+
+class Serializable(ABC):
+    _registry = None
+
+    @staticmethod
+    def registry():
+        if Serializable._registry is None:
+            Serializable._registry = {c.serializer_name(): c for c in Serializable.__subclasses__()}
+        return Serializable._registry
+
+    @abstractmethod
+    def _serialize(self):
+        """Return a plain python datatype which is json serializable."""
+        pass
+
+    def serialize(self):
```

Comment:
May I be so bold to suggest adding some return types? Especially for public api methods like this

Reply:
The return type is 'any json serializable type', which I wrote in the docstring, but am not sure how better to document.